internal/gps/source_manager.go

```diff
@@ -349,22 +332,13 @@ func (e CouldNotCreateLockError) Error() string {
 // longer safe to call methods against it; all method calls will immediately
 // result in errors.
 func (sm *SourceMgr) Release() {
-	// Set sm.releasing before entering the Once func to guarantee that no
-	// _more_ method calls will stack up if/while waiting.
-	atomic.CompareAndSwapInt32(&sm.releasing, 0, 1)
-
-	// Whether 'releasing' is set or not, we don't want this function to return
```

Comment:
ahh - this is the key, and why we need the `sync.Once()`. the proposed change would cause `Release()` to return immediately, even if the real work of `doRelease()` hasn't completed yet. so: say a signal is sent, which causes a call made by the solver to fail, causing the solver to error out. at that point, it's a very direct path to the entire dep process exiting. the only thing that prevents it from doing so is the call it makes to `Release()`. with this change, it would exit right away, as the signal handler would have already flipped the `sm.releasing` flag.

this could, albeit quite unlikely, result in some inconsistent disk state with the persistent cache, which @jmank88 is currently working on implementing.

i suspect the `SourceMgr` is probably strictly crash-only right now, but that's not currently an explicit design goal. items in the persistent cache have dependences on one another, and we don't necessarily impose any kind topological write order in a way that guarantees dependencies enter the cache prior to their dependers. we are, in general, self-healing in the face of such absent dependent data, as there are circumstances where it can occur during normal operation. now, it would certainly be wise to achieve such an ordered write property, as it would more properly insulate us against true system failure.

but, at least until we make than an explicit goal, allowing any possibility of any call to `Release()` to return before all resources have actually released seems like tempting fate.

as a more practical matter, we also want to be very sure that there is _no_ way any subprocesses can still be running after any call to `Release()` returns. we had scads of problems early on in our tests where unfinished e.g. git subprocesses, especially on windows, were causing intermittent test failures due to file locks they held on directories that _should_ have been fine to remove.

Reply:
probably worth updating the documentation to note this property about after-return-time, in addition to what it already says about after-call-time.

Reply:
OK, so you want calls to Release to wait, is that it? Done.